src/portfft/descriptor_validate.hpp

```diff
@@ -0,0 +1,217 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's portFFT
+ *
+ **************************************************************************/
+
+#ifndef PORTFFT_DESCRIPTOR_VALIDATE_HPP
+#define PORTFFT_DESCRIPTOR_VALIDATE_HPP
+
+#include <string_view>
+
+#include "common/exceptions.hpp"
+#include "enums.hpp"
+#include "utils.hpp"
+
+namespace portfft::detail::validate {
+
+inline void validate_lengths(const std::vector<std::size_t>& lengths) {
+  if (lengths.empty()) {
+    throw invalid_configuration("Invalid lengths, must have at least 1 dimension");
+  }
+  for (std::size_t i = 0; i < lengths.size(); ++i) {
+    if (lengths[i] == 0) {
+      throw invalid_configuration("Invalid lengths[", i, "]=", lengths[i], ", must be positive");
+    }
+  }
+}
```

Comment:
This function is called from the constructor of `committed_descriptor`, before `build_w_spec_const`. I put all the validation stuff in a separate file because it can easily be compartmentalised in its own little place, and not distract from other bits.

Reply:
Its perfectly alright where the functions are defined. Its about where they are called from.  Combining this with [this comment](https://github.com/codeplaysoftware/portFFT/pull/136#discussion_r1472731845), what I am trying to discuss, is that, since a user would specify the layout configuration in the descriptor itself, should we let someone commit the descriptor at all, if the provided layout configuration is invalid

Reply:
Another thing to consider over here might be how it would interact with our oneMKL integration

Reply:
So are you suggesting that the call to `detail::validate::validate_descriptor` should be [here](https://github.com/codeplaysoftware/portFFT/blob/027cc4189beedb2fa756bd1ecd1b8ff6cea189e6/src/portfft/descriptor.hpp#L1606) instead?

Reply:
yes, I do not suppose we should reach the constructor of committed_descriptor if the layout is illegal

Reply:
Sounds pretty reasonable to me. https://github.com/codeplaysoftware/portFFT/pull/136/commits/f54b9b773e5e54de69e3eaffc109f68b42a8be33